arch/x86/chunkset_avx.c

```diff
@@ -14,6 +15,43 @@ typedef __m256i chunk_t;
 #define HAVE_CHUNKMEMSET_2
 #define HAVE_CHUNKMEMSET_4
 #define HAVE_CHUNKMEMSET_8
+#define HAVE_CHUNK_MAG
+
+/* Populate don't cares so that this is a direct lookup 
+ * (with some indirection into the permute table), because
+ * dist can never be 0 - 2, we'll start with an offset,
+ * subtracting 3 from the input */
+static const lut_rem_pair perm_idx_lut[29] = {
+    {0, 2},                 /* 3 */
+    {0, 0},                 /* don't care */
+    {1 * 32, 2},            /* 5 */
+    {2 * 32, 2},            /* 6 */
+    {3 * 32, 4},            /* 7 */
+    {0 * 32, 0},            /* don't care */
+    {4 * 32, 5},            /* 9 */
+    {5 * 32, 22},           /* 10 */
+    {6 * 32, 21},           /* 11 */
+    {7 * 32, 20},           /* 12 */
+    {8 * 32, 16},           /* 13 */
+    {9 * 32, 4},            /* 14 */
+    {10 * 32, 2},           /* 15 */
+    {0 * 32 , 0},           /* don't care */
```

Comment:
Space before comma looks weird... Maybe should use space before `0` instead and use ` 0 * 32` consistently to make second item of array element align better... Also add space before `1` ... `9` when the sequence goes higher than `9`.